File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minP = prices[0]
        maxProfit = 0
        for p in prices:
            if p < minP:
                minP = p
            currentProfit = p - minP
            maxProfit = max(maxProfit, currentProfit)
        return maxProfit
       

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int buyPrice = prices[0];
        for (int p : prices) {
            if (p > buyPrice) {
                maxProfit = Math.max(maxProfit, p - buyPrice);
            } else {
                buyPrice = p;
            }
        }
        return maxProfit;
    }
}